### DISCUSSION TITLE
Can't update firmware : fwupdx64.efi.signed missing

### DISCUSSION THREAD

**Discussion by gbdomubpkm** [fwupd/fwupd#3688] [Q&A] [answered]
Device 

LENOVO ThinkPad X280 with Ubuntu 20.04.3 LTS installed

![15082021 ThinkPad X280 fwupdx64 efi signed missing can't update firmware](https://user-images.githubusercontent.com/50362604/131212826-a9a4444f-df01-4fb8-a96e-969716ced06c.png)

Hello
First i'm not a developer, only a user with a basic knowledge, so forgive me if i'm not precise enough..

Two weeks ago i think, i received these update in SNAP store and i tried to upgrade firmware with snap store but not possible to upgrade.
Then by going on forums i tried the command line   **sudo fwupdmgr update**   if i remember correctly : not possible too because of the signed file missing. I also had to send a bug report (is it to Ubuntu or Lenovo ?) at the end of the unsuccessful command.
Still unable to update to date. Should I wait for the snap store to fix the problem?
I would like to point out this problem if you can solve it (easily) or if you can pass on the information to the right hands.

With kind regards

**Comment by superm1** [comment]
It sounds like you're missing the package `fwupd-signed` on your system.  Can you install it?

**Comment by gbdomubpkm** [comment]
Yes indeed, i'm missing this signed package (issue title).
How to install this signed package ? I don't know how to do that.
Is this the right approach? If so, do you have a complete procedure? Thank you in advance.

**Comment by superm1** [comment, marked as answer]
`apt install fwupd-signed`

**Reply by gbdomubpkm** [reply]
Only a sudo before and it is solved ! 🙂 Packages are upgraded too. Thank you very much !
